src/plugins/hetero/plugin.cpp

```diff
@@ -72,36 +73,36 @@ std::string Engine::GetTargetFallback(const Engine::Configs& config, bool raise_
 }
 
 InferenceEngine::IExecutableNetworkInternal::Ptr Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork& network,
-                                                                            const Configs& config) {
+                                                                            const Configs& user_config) {
     if (GetCore() == nullptr) {
         IE_THROW() << "Please, work with HETERO device via InferencEngine::Core object";
     }
-    auto tconfig = mergeConfigs(_config, config);
-    std::string fallbackDevicesStr = GetTargetFallback(tconfig);
-    DeviceMetaInformationMap metaDevices = GetDevicePlugins(fallbackDevicesStr, tconfig);
+    auto full_config = mergeConfigs(_config, user_config);
+    std::string fallbackDevicesStr = GetTargetFallback(full_config);
+    DeviceMetaInformationMap metaDevices = GetDevicePlugins(fallbackDevicesStr, user_config);
```

Comment:
@songbell note, that here (and in auto batch) I changed full_config to user_config when I pass values to Core::get_supported_property. The same should be done for AUTO, because user-set values should not be merged with global plugin state.

Merging of user and global plugin properties is needed only for virtual plugin properties, but not for underlying plugins.

Reply:
probably applicable for batch and hetero, but currently for auto, its properties including 2 parts: self properties for internal state control + properties that we expect user can set to auto (and then merged with user config for compile model), like perfhint, perfcount etc.
something like core.set_property("AUTO", ov::hint::performance_mode("THROUGHPUT"), ov::execution_mode("PERFORMANCE")));
then user don't have to designate these params in general compile_model.

Reply:
@wangleis please note about the possible user scenario change here

Reply:
Ok, agree.
Then we need to separate user options and internal options (as far as I remember, some sort of separation is done in AUTO). And user options should not have default values (map of options should be empty).

In this case global_user_options (set in set_property) and user_options (set in compile / query model) should be merged.
If will allow to remove checks in auto like `if_set_by_user`

Reply:
@songbell @ilya-lavrenov  I proposed new approach at https://github.com/openvinotoolkit/openvino/pull/16489#discussion_r1147097733 which also cover this part. Please take a look and comment.